lib/Conversion/TorchToLinalg/DataMovement.cpp

```diff
@@ -135,30 +135,24 @@ class ConvertAtenViewOp : public OpConversionPattern<AtenViewOp> {
           op, "desired size list length mismatches with the result type rank");
     }
 
+    // If all values in the shape are known, then lower to `tensor::ReshapeOp`.
+    ArrayRef<int64_t> outShape = resultType.getShape();
+    if (llvm::none_of(outShape, [](int64_t shape) { return shape == -1; })) {
+      SmallVector<Value> outShapeValues =
+          getAsConstantIndexValues(rewriter, loc, outShape);
+      auto shapeTensor =
+          rewriter.create<tensor::FromElementsOp>(loc, outShapeValues);
+      rewriter.replaceOpWithNewOp<tensor::ReshapeOp>(op, resultType, input,
+                                                     shapeTensor);
+      return success();
+    }
+
     SmallVector<Value> inputSize = getTensorSizes(rewriter, loc, input);
     ArrayRef<Value> expandedShapeInt =
         llvm::makeArrayRef(isCollapse ? inputSize : outputSizeInt);
     ArrayRef<Value> collapsedShapeInt =
         llvm::makeArrayRef(isCollapse ? outputSizeInt : inputSize);
 
-    // Currently, we only handle the expanding or collapsing cases or the
-    // identity cases where the rank and shape of the input and result are
-    // equal, and the input itself is the result. We do not handle expanding And
-    // collapsing happening at the same time or cases where it's neither
-    // collapsing nor expanding like view of [2,3] for 3x2 tensor.
-    // TODO: For the expanding And collapsing case, we will need to identify
-    // which dimensions are collapsing and which are expanding and do it in two
-    // steps.
-    // TODO: For neither collapsing nor expanding, we could find a intermediate
-    // shape to collapse and then expanded to the target shape. Like [2,3] =>
-    // [6] => [3, 2].
-    if (inputRank == resultRank) {
```

Comment:
I think there is still value to keeping this here, since it handles some cases that are not handled by the `ReshapeOp`, namely, the case where sizes are dynamic. I would just update the comment to mention that once `ReshapeOp` handles dynamic sizes, if ever, then this will no longer be needed.